packages/expo-modules-core/android/src/main/cpp/types/FrontendConverter.cpp

```diff
@@ -40,6 +40,23 @@ bool IntegerFrontendConverter::canConvert(jsi::Runtime &rt, const jsi::Value &va
   return value.isNumber();
 }
 
+jobject LongFrontendConverter::convert(
+  jsi::Runtime &rt,
+  JNIEnv *env,
+  JSIInteropModuleRegistry *moduleRegistry,
+  const jsi::Value &value
+) const {
+  auto &longClass = JavaReferencesCache::instance()
+    ->getJClass("java/lang/Long");
+  jmethodID longConstructor = longClass.getMethod("<init>", "(J)V");
+  return env->NewObject(longClass.clazz, longConstructor,
+                        static_cast<jlong>(value.getNumber()));
+}
+
+bool LongFrontendConverter::canConvert(jsi::Runtime &rt, const jsi::Value &value) const {
+  return value.isNumber();
```

Comment:
having a question here, maybe also for the integer converter. what will happen if the js value is `3.14` (a decimal number)

Reply:
The decimal part will be ignored in that case. We can't do much here. I don't want to throw an error in that case 🤷

Reply:
that makes sense. thanks for sharing the info.